tasks/mynt.ts

```diff
@@ -965,14 +981,115 @@ task(
         [basketManagerProxy.address, newBasketManagerImpl.address]
       );
 
-      const sipArgs: ISipArgument = {
+      const sipArgs: ISipArgument["args"] = {
         targets: [basketManagerProxy.address],
         values: [0],
         signatures: [signatureUpgrade],
         data: [dataUpgrade],
         description: "Upgrade fees vault contract",
       };
-
-      createSIP(hre, sipArgs);
+      logger.warn(">>> CREATE A SIP WITH THIS ARGS: <<<");
+      logger.info(sipArgs);
+      logger.warn("====================================");
     }
   });
+
+task("upgrade:mocIntegration", "Upgrade implementation of mocIntegration contract")
+.addOptionalParam("isMultisig", "flag if transaction needs to be intiated from the multisig contract")
+.addOptionalParam("isSIP", "flag if transaction needs to be initiated from the SIP")
+.setAction(async ({ isMultisig, isSIP }, hre) => {
+  helpers.injectHre(hre);
+  const { ethers, deployments: { get }, getNamedAccounts } = hre;
+  const { deployer } = await getNamedAccounts();
+  const myntAdminProxy = await ethers.getContract("MyntAdminProxy");
+  const mocIntegrationProxy = await ethers.getContract("MocIntegration"); // MocIntegration
+
+  const MocIntegrationFactory = await ethers.getContractFactory("MocIntegration");
+  const newMocIntegrationImpl = await MocIntegrationFactory.deploy();
+  console.log(`Upgrading mocIntegration implementation to ${newMocIntegrationImpl.address}`)
+
+  if(isMultisig) {
+    const multisigAddress = (await get("MultisigWallet")).address;
+    const dataUpgrade = myntAdminProxy.interface.encodeFunctionData("upgrade", [
+      mocIntegrationProxy.address, newMocIntegrationImpl.address
+    ]);
+
+    await helpers.sendWithMultisig(
+      multisigAddress,
+      myntAdminProxy.address,
+      dataUpgrade,
+      deployer
+    );
+  } else if(isSIP) {
+    const signatureUpgrade = "upgrade(address,address)";
+    const dataUpgrade = myntAdminProxy.interface.encodeFunctionData("upgrade", [
+      mocIntegrationProxy.address, newMocIntegrationImpl.address
+    ]);
+
+    const sipArgs: ISipArgument = {
+      targets: [mocIntegrationProxy.address],
+      values: [0],
+      signatures: [signatureUpgrade],
+      data: [dataUpgrade],
+      description: "Upgrade MocIntegration contract"
+    }
+
+    createSIP(hre, sipArgs);
```

Comment:
we can't create a SIP like this - most probably it will be a multiple upgrades, but can print params for the SIP, but i recommend to include it into the deployment script.

Reply:
updated in this [commit](https://github.com/DistributedCollective/mynt/pull/17/commits/57307ba37bc62d3f003c1b59b86350579a01d092)